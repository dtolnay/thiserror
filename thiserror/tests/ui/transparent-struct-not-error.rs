use thiserror::Error;

#[derive(Error, Debug)]
#[error(transparent)]
pub struct Error {
    message: String,
}

fn main() {}
