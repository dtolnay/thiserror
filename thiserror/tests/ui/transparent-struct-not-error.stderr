error[E0599]: the method `as_dyn_error` exists for struct `String`, but its trait bounds were not satisfied
 --> tests/ui/transparent-struct-not-error.rs:4:9
  |
4 | #[error(transparent)]
  |         ^^^^^^^^^^^ method cannot be called on `String` due to unsatisfied trait bounds
  |
 ::: $RUST/alloc/src/string.rs
  |
  | pub struct String {
  | ----------------- doesn't satisfy `String: AsDynError<'_>` or `String: std::error::Error`
  |
  = note: the following trait bounds were not satisfied:
          `String: std::error::Error`
          which is required by `String: AsDynError<'_>`
          `str: Sized`
          which is required by `str: AsDynError<'_>`
          `str: std::error::Error`
          which is required by `str: AsDynError<'_>`
